install.ps1

```diff
@@ -0,0 +1,601 @@
+    # try for zst first; if that fails use gz for older release branches
+    if (!(Test-Download -Uri $AirgapChecksumsUrl)) {
+        $AirgapChecksumsUrl = "$StorageUrl/rke2-images.$suffix$CommitHash.tar.gz.sha256sum"
+    }
+    Write-InfoLog "downloading airgap checksums at $AirgapChecksumsUrl"
+    Invoke-RestMethod -Uri $AirgapChecksumsUrl -OutFile $-TempAirgapChecksums
+    return Find-Checksum -Path $-TempAirgapChecksums -Pattern "rke2-images.$suffix.tar"
+}
+
+# download_airgap_tarball downloads the airgap image tarball.
+function Get-AirgapTarball() {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [String]
+        $CommitHash,
+        [Parameter()]
+        [String]
+        $AirgapTarballUrl,
+        [Parameter()]
+        [String]
+        $StorageUrl,
+        [Parameter()]
+        [String]
+        $TempAirgapTarball        
+    )
+
+    if ($CommitHash){
+        return
+    }    
+
+    $archInfo = Get-ArchitectureInfo
+    $suffix = $archInfo.Suffix    
+
+    $AirgapTarballUrl= "$StorageUrl/rke2-images.$suffix$CommitHash.tar.zst"
+
+    # try for zst first; if that fails use gz for older release branches
+    if (!(Test-Download -Url $AirgapTarballUrl)) {
+        $AirgapTarballUrl = "$StorageUrl/rke2-images.$suffix$CommitHash.tar.gz"
+    }
+    Write-InfoLog "downloading airgap tarball at $AirgapTarballUrl"
+    Invoke-RestMethod -Uri $AirgapTarballUrl -OutFile $TempAirgapTarball
+}
+
+# verify_airgap_tarball compares the airgap image tarball checksum to the value
+# calculated by CI when the file was uploaded.
+function Test-AirgapTarballChecksum() {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [String]
+        $ExpectedAirGapChecksum,
+        [Parameter()]
+        [String]
+        $TempAirGapTarball
+    )
+
+    if ($ExpectedAirGapChecksum) {
+        return
+    }
+    Write-InfoLog "verifying airgap tarball"
+    $actualAirgapChecksum = $(sha256sum "$TempAirGapTarball" | awk '{print $1}')
+    if ($ExpectedAirGapChecksum -ne $actualAirgapChecksum) {
+        Write-FatalLog "download sha256 does not match $ExpectedAirGapChecksum, got $actualAirgapChecksum"
+    }
+}
+
+# install_airgap_tarball moves the airgap image tarball into place.
+function Install-AirgapTarball() {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [String]
+        $InstallAgentImageDir,
+        [Parameter()]
+        [String]
+        $TempAirgapTarball,
+        [Parameter()]
+        [String]
+        $ExpectedAirGapChecksum,
+        [Parameter()]
+        [String]
+        $AirgapTarballFormat,
+        [Parameter()]
+        [String]
+        $TempAirgapChecksums
+    )    
+
+    if ($ExpectedAirGapChecksum) {
+        return
+    }
+    New-Item -Path "$InstallAgentImageDir" -ItemType "Directory"
+    $archInfo = Get-ArchitectureInfo
+    $suffix = $archInfo.Suffix
+
+    # releases that provide zst artifacts can read from the compressed archive; older releases
```

Comment:
The old non-zst-supporting releases won't have Windows builds, so you don't need to worry about this. Just use zst for everything.

Reply:
Good call, removed.